ODIN_II/SRC/netlist_create_from_ast.cpp

```diff
@@ -1925,12 +1926,15 @@ void connect_module_instantiation_and_alias(short PASS, ast_node_t* module_insta
                      * already been instantiated without any initial value. */
                     nnet_t* output_net = (nnet_t*)output_nets_sc->data[sc_spot_output];
                     nnet_t* input_new_net = (nnet_t*)output_nets_sc->data[sc_spot_input_new];
-                    if (output_net->driver_pin && output_net->driver_pin->node && input_new_net) {
-                        output_net->driver_pin->node->initial_value = input_new_net->initial_value;
+                    for (int k = 0; k < output_net->num_driver_pins; k++) {
+                        // TODO I am unsure if this is the correct behaviour
```

Comment:
@jeanlego I cant figure out what this code is trying to achieve so I cant say whether this is correct. Do you have any insight? The current behaviour is my best guess but I don't understand what scenario we would want to set all driver's initial values.

Reply:
So nets technically don't have initial values, only latches can hold one. But when we build the circuit, we assign the init value to the net., With that in mind, we need to push that init value down to the node that the net is driving, of there is an init value on the net other then unknown, you should expect a latch bellow

Reply:
So the driver should be a latch? In which case we would expect this to only apply when there one driver, else it would be an error since the net has multiple drivers one of which is a latch that is not guarded by a bufif/MUX

Reply:
No, the driver you are looking at is driving a latch, the init value just hasn't been pushed down on the latch yet

Reply:
From looking into it I can see a situation that we have multiple drivers one of which is a latch. There are cases were I can see it being valid, for example constant conditions or preprocessor behaviour that causes only one driver to resolve to connected. In these cases we need to propagate the initial value to whichever driver is the latch. I have added asserts elsewhere in the pipeline that should catch the case where two distinct initial values (not including Z) are associated with the net. This case should always be ill formed for our purposes to the best of my knowledge.